sda-download/api/sda/sda.go

```diff
@@ -272,23 +272,18 @@ func Download(c *gin.Context) {
 		return
 	}
 
-	switch c.Param("type") {
-	case "encrypted":
-		// calculate coordinates
-		start, end, err = calculateEncryptedCoords(start, end, c.GetHeader("Range"), fileDetails)
-		if err != nil {
-			log.Errorf("Byte range coordinates invalid! %v", err)
+	wholeFile := true
+	if start != 0 || end != 0 {
+		wholeFile = false
+	}
 
-			return
-		}
-		if start > 0 {
-			// reading from an offset in encrypted file is not yet supported
-			log.Errorf("Start coordinate for encrypted files not implemented! %v", start)
-			c.String(http.StatusBadRequest, "Start coordinate for encrypted files not implemented!")
+	start, end, err = calculateCoords(start, end, c.GetHeader("Range"), fileDetails, c.Param("type"))
```

Comment:
If you are refactoring coordinate/range handling already, I think a cleaner interface would be passing the context itself, so ideally for me this would be something like

```
start, end, rangeSpecified, err := calculateCoords(c)
```

to pick up either range or startCoordinate and endCoordinate. I strongly think we want this harmonized for all use cases (I definitely want Range supported for both download without s3 and unencrypted s3).

I think it might make somewhat more sense to have a separate function to transform the coordinates from the underlying data stream coordinate system to encrypted coordinates and having header adjustments at the time of need (`Content-length` I assume).

Reply:
I do agree, it would be very nice to refactor this. But in order to get this merged before the deadline, I wonder if it would not make sense to save all the refactoring for later. Considering the risk of introducing new errors and the time we need to spend testing after refactoring...
Maybe create a new issue for this and list all ideas there?

Reply:
to be picked up in #834 .